include/grpc++/auth_context.h

```diff
@@ -87,6 +87,11 @@ class AuthContext {
   // Iteration over all the properties.
   virtual AuthPropertyIterator begin() const = 0;
   virtual AuthPropertyIterator end() const = 0;
+
+  // Mutation functions: should only be used by an AuthMetadataProcessor.
+  virtual void AddProperty(const grpc::string& key,
```

Comment:
You mean a property right?
So, yes, this is the intention. You cannot remove/replace properties from the passed-in auth context (which basically comes from the channel and is chained at the lowest layer). This is Add only.